### PR TITLE
AI-849: Newly added location is not available in list of choices for Site Type

### PR DESCRIPTION
**Pull request by bddbot**
[AI-849](https://bedatadriven.atlassian.net/browse/AI-849): Newly added location is not available in list of choices for Site Type

1. Navigate to the design page for a database with design privileges
2. Create a new Location type called "ClosedLocationType" and select "Users may add new locations during data entry"
3. Now create a new Form called "ClosedLocationForm" and choose "Classic" mode.

Current Result: You will note that the 'ClosedLocationType' won't be available in list of choices for Site Type: http://www.screencast.com/t/JNJZeHI6giAz
 
Expected Result: "CLosedLocationType" should appear in the list of choices for "Site Type"

### CODE REVIEW

File: server/src/main/java/org/activityinfo/ui/client/page/config/design/ActivityForm.java [hunk 1]
```diff
@@ -77,9 +80,17 @@ public ActivityForm(Dispatcher service, UserDatabaseDTO database) {
         add(categoryField);
 
         final MappingComboBox<Integer> locationTypeCombo = new MappingComboBox<Integer>();
-        for (LocationTypeDTO type : database.getCountry().getLocationTypes()) {
-            locationTypeCombo.add(type.getId(), type.getName());
-        }
+        database.then(new Function<UserDatabaseDTO, Object>() {
```
Comment (akbertram) [inline server/src/main/java/org/activityinfo/ui/client/page/config/design/ActivityForm.java]: What happens if the request fails?
Whenever a combo box is populated by a remote source, we need to address the case where the request fails. If the request fails here, then the form will be in invalid state and the user will be confused.

This is the same problem we have with the GWTChosen widget, but here at least with GXT we have the right component: it should be RemoteComboBox with an asynchronous loader.

Comment (yuriyz) [inline server/src/main/java/org/activityinfo/ui/client/page/config/design/ActivityForm.java]: @akbertram used RemoteComboBox to cover request failure in https://github.com/bedatadriven/activityinfo/commit/11ed459909fb70d0f5c5970e01563f8cfb91dc55

**Comment by akbertram** [comment]
@bddbot please release

**Comment by akbertram** [comment]
Already fixed.

**Pull request closed without merging**